# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/base/content/pageinfo/security.js
# Commit: 11e5baee978e
# Full Hash: 11e5baee978e8f98fa3795e430dd60c7600c9e8c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1580315
# File Overlap Count: 1
# Description:
#   Bug 1580315 - Convert certList to raw array for nsITransportSecurityInfo r=keeler,Ehsan,kershaw
#   
#   This patch converts the certList attribute of nsITransportSecurityInfo
#   from nsIX509CertList to Array<nsIx509Cert>
#   
# ==============================================================================

diff -r 341d72ef4bf5 -r 11e5baee978e browser/base/content/pageinfo/security.js
--- a/browser/base/content/pageinfo/security.js	Mon Oct 21 19:33:02 2019 +0000
+++ b/browser/base/content/pageinfo/security.js	Mon Oct 21 19:49:01 2019 +0000
@@ -32,7 +32,7 @@
 
   viewCert() {
     if (Services.prefs.getBoolPref("security.aboutcertificate.enabled")) {
-      let certChain = getCertificateChain(this.securityInfo.certChain);
+      let certChain = this.securityInfo.certChain;
       let certs = certChain.map(elem =>
         encodeURIComponent(elem.getBase64DERString())
       );
@@ -388,14 +388,6 @@
   }
 }
 
-function getCertificateChain(certChain, options = {}) {
-  let certificates = [];
-  for (let cert of certChain.getEnumerator()) {
-    certificates.push(cert);
-  }
-  return certificates;
-}
-
 /**
  * Return true iff realm (proto://host:port) (extracted from uri) has
  * saved passwords